test/mechanic_machine_test.rb

```diff
@@ -172,4 +172,64 @@ def test_error_raised_when_states_are_duplicated_with_normal_enum_entry
       end
     end
   end
+
+  if Rails::VERSION::STRING >= '5'
+    def test_enum_definition_with_prefix
+      ActiveRecord::Migration.create_table(:enum_prefix_test) do |t|
+        t.integer :status
+        t.integer :comments_status
+      end
+      tes = Class.new(ActiveRecord::Base) do
+        self.table_name = 'enum_prefix_test'
+        enum(status: [:active, :archived], _prefix: true) { event(:archive) { transition(active: :archived) } }
+        enum(comments_status: [:active, :inactive], _prefix: :comments) { event(:close) { transition(active: :inactive) } }
+      end.new status: :active, comments_status: :active
+
+      assert({ active: :archived }, tes.status_archive_transition)
```

Comment:
I think `assert_equal` is appropriate.
 
`Minitest::Assertions#assert(test, msg = nil)` check if
first argument (`test`) is not falsy value, and second argument (`msg`)
is used as a custom error message.
`assert({ active: :archived }, tes.status_archive_transition)` always passes because first arguemtn (`{ active: :archived }`) is truthy.